src/index.ts

```diff
@@ -84,7 +84,7 @@ function parseXPath(xpathExpression: EvaluableExpression): Expression {
 
 	const ast =
 		typeof xpathExpression === 'string'
-			? parseExpression(xpathExpression, { allowXQuery: false })
+			? parseExpression(xpathExpression, { allowXQuery: false, version: 4 })
```

Comment:
Should this depend on the options?

In that case we'd probably also need separate caches for the 3.1 vs 4 expressions.